### PR TITLE
Feature etb#195 uart hwas c driver

### PR DESCRIPTION
**Pull request by jrachucki-n7s**

### CODE REVIEW

File: TestUtils/main_FreeRTOS.h [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+{
+  (void)pxTask;
+  (void)pcTaskName;
+
+  configASSERT(0);
+}
+
+void vApplicationIdleHook()
+{
+  volatile size_t xFreeHeapSpace = xPortGetFreeHeapSize();
```
Comment (Lurkerpas) [inline TestUtils/main_FreeRTOS.h]: Why is this needed?

File: Drivers/c/UartHwas/test/main.cpp [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+  config.id = UartHwas_Id_0;
+
+  //< Exercise
+  UartHwas_init(&uart, &config);
+
+  //< Verify
+  VerifyInit(uart, config);
+}
+
+/// \Given uninitialized UAbr U/// \When UART1 is initialized with 9600 bps
```
Comment (Lurkerpas) [inline Drivers/c/UartHwas/test/main.cpp]: newline?

File: Drivers/c/UartHwas/test/main.cpp [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+
+  //< Exercise
+  UartHwas_init(&uart, &config);
+
+  //< Verify
+  VerifyInit(uart, config);
+}
+
+/// \Given uninitialized UART
+/// \When UART0 is initialized with 115200 bps
```
Comment (Lurkerpas) [inline Drivers/c/UartHwas/test/main.cpp:242]: UART0 vs UartHwas_Id_4. 
On a side note, this piece of code does not require exhaustive testing of all possible Uarts..... Testing the one target config, or the two possible target configs (the ones connected to the actual board) should be enough.

Comment (Lurkerpas) [reply Drivers/c/UartHwas/test/main.cpp:242]: But again, if you want to test multiple configs, it is good to just make a loop...

File: TestUtils/main_FreeRTOS.h (hunk 1 above)
Comment (jrachucki-n7s) [reply TestUtils/main_FreeRTOS.h]: It is not needed. I have deleted that in new commit

File: Drivers/c/UartHwas/test/main.cpp (hunk 3 above)
Comment (jrachucki-n7s) [reply Drivers/c/UartHwas/test/main.cpp:242]: Changed to loop in new commit

**Review by Lurkerpas — APPROVED** [review]
